Feature: Add disable prop

oieduardorabelo opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**What kind of change does this PR introduce?**

- [ ] bugfix
- [x] feature
- [ ] refactoring / style
- [ ] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [x] Yes, my code is well tested
- [ ] Not relevant

**If relevant, did you update the documentation?**

- [x] Yes, I've updated the documentation
- [ ] Not relevant

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->
feat request in https://github.com/okonet/react-dropzone/issues/472, be able to  to disable the dropzone entirely

hey @okonet , just one idea to implement #472

okonet commented on src/index.js:
```diff
@@ -410,6 +424,11 @@ Dropzone.propTypes = {
   /**
+ * Enable/disable the dropzone entirely
+ */
+  disableDropzone: PropTypes.bool,
```
Let's rename the prop to `disabled` to conform the DOM API: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

okonet commented on src/index.js:
```diff
@@ -344,7 +356,8 @@ class Dropzone extends React.Component {
       ref: this.setRefs,
-      onChange: this.onDrop
+      onChange: this.onDrop,
+      disabled: disableDropzone
```
After the prop is renamed, let's put it after the `accept` one using the shorthand notation.

okonet commented on src/index.js:
```diff
@@ -66,6 +69,14 @@ class Dropzone extends React.Component {
     document.body.onfocus = null
   }
 
+  composeHandlers(handler, target) {
```
Since this is only called in the constructor and it returns `null`, this won't react to the change of the prop. We'll need to add a test for this (change from disabled to enabled state and check that events are being fired).

okonet commented:
Thanks for working on it!

jacksonrayhamilton commented:
Thanks @oieduardorabelo for adding this!  And just in time, as our product just started needing this functionality, as well.  :smile:

oieduardorabelo commented:
hey @okonet,

- I've moved the compose fn to the render and added tests for the react lifecycle updates
- I also exposed `disabledStyle` and `disabledClassName`, you know, stylesss! (maybe the default `opacity` isn't a good idea? anyway, up to you)

jacksonrayhamilton commented:
@okonet Can we expect a release with this feature soon?

okonet commented:
See https://github.com/react-dropzone/react-dropzone/releases/tag/v4.1.0

It wasn't in /releases since I've moved the repo to another organization and some semantic-release functions were fucked up.

jacksonrayhamilton commented:
Ah, missed that.  Thanks!
